### PR TITLE
Pass through octal and binary literals as-is

### PR DESCRIPTION
**Pull request by JimPanic**
See https://github.com/coffeescript6/discuss/issues/45

Note: I'm deliberately creating this PR towards branch 2 because it is not opt-in and I'm not sure how many current JS environments implement the binary and octal literals as ES2015 does.

### CODE REVIEW

File: src/lexer.coffee [hunk 1]
```diff
@@ -202,14 +204,12 @@ exports.Lexer = class Lexer
...
-      numberValue = parseInt(octalLiteral[1], 8)
-      number = "0x#{numberValue.toString 16}"
-    else if binaryLiteral = /^0b([01]+)/.exec number
-      numberValue = parseInt(binaryLiteral[1], 2)
-      number = "0x#{numberValue.toString 16}"
-    else
+
+    numberValue = number
+
+    unless /^(0o[0-7]+)/.test(number) and /^(0b[01]+)/.test(number)
```
Comment (lydell) [inline src/lexer.coffee]: Isn’t this a constant condition (`unless false`)?

File: test/numbers.coffee [hunk 2]
```diff
@@ -13,9 +13,26 @@
 # Binary Integer Literals
 # Binary notation is understood as would be decimal notation.
 
+toJS = (str) ->
+  CoffeeScript.compile str, bare: yes
+  .replace /^\s+|\s+$/g, '' # Trim leading/trailing whitespace
+
```
Comment (lydell) [inline test/numbers.coffee]: Following the style of all other tests (except import/export), all these added tests should be removed.

Comment (JimPanic) [inline test/numbers.coffee]: Oh, I didn't know that. So no compilation tests at all?

File: src/lexer.coffee (hunk 1 above)
Comment (JimPanic) [inline src/lexer.coffee]: Good catch, thanks.

File: test/numbers.coffee (hunk 2 above)
Comment (lydell) [inline test/numbers.coffee]: Yeah, that’s been the rule. Only check that the CoffeeScript does what is expected.

Comment (JimPanic) [inline test/numbers.coffee]: Ok, cool. 👍 I like removing things.

**Comment by JimPanic** [comment]
Removed the tests and fixed the logic.

**Comment by lydell** [comment]
@JimPanic While reviewing this PR, I found a bug in master. I fixed that bug, and have also updated the 2 branch with the fix. Could you please rebase this PR?

**Comment by JimPanic** [comment]
Just to be clear: is there a reason to parse hex, oct and binary in `2`?

**Comment by JimPanic** [comment]
Oh, of course they need to be parsed - for the Infinity check.

**Comment by lydell** [comment]
Yep. Btw, you seem to have completely messed up the commits now :) git is so much fun

**Comment by JimPanic** [comment]
Yes, totally. And your fix confused me. :) Basically you refactored it so well, I just have to remove two lines again. So this will be a new force push to my branch; on top of your last commit on 2.

**Comment by lydell** [comment]
Nice! LGTM. @GeoffreyBooth what about you?

**Comment by GeoffreyBooth** [comment]
Is this really just +1 −2 changes? It doesn’t get much simpler than that. Looks good to me assuming we don’t need any tests (?)

**Comment by lydell** [comment]
I've missed compiled output tests from time to time. For example, when reviewing this PR, I found an output bug that still gave the same runtime result. It would be nice to write tests for such things.

**Comment by JimPanic** [comment]
Hence why I added the tests in the first place, but I didn't know there was this rule.

We could have compilation tests in a seperate folder/repository maybe?

**Comment by lydell** [comment]
Compilation tests shouldn't be a blocker for this PR, though? That's a separate issue?

**Comment by JimPanic** [comment]
Yeah, that's definitely something for a different issue.

**Pull request merged**